dace/codegen/targets/unroller.py

```diff
@@ -23,31 +31,111 @@ def __init__(self, frame_codegen: DaCeCodeGenerator, sdfg: dace.SDFG):
         dispatcher = self._dispatcher
 
         # Register dispatchers
-        self._dispatcher.register_map_dispatcher(dace.ScheduleType.Unrolled, self)
+        self._dispatcher.register_map_dispatcher(dace.ScheduleType.Unrolled,
+                                                 self)
 
     def get_generated_codeobjects(self):
         return []
 
+    #TODO: Expand the unroller so it can also generate openCL code
     def generate_scope(self, sdfg: dace.SDFG, scope: ScopeSubgraphView,
                        state_id: int, function_stream: CodeIOStream,
                        callsite_stream: CodeIOStream):
-        
-        entry_node = scope.source_nodes()[0]
 
+        entry_node = scope.source_nodes()[0]
         index_list = []
+
+        #Generate new names for nsdfgs, and adds defined variables to constants
+        def nsdfg_prepare_unroll(scope, paramname, paramval):
+            unique_functions_conf = Config.get('compiler', 'unique_functions')
+            if unique_functions_conf is True:
+                unique_functions_conf = 'hash'
+            elif unique_functions_conf is False:
+                unique_functions_conf = 'none'
+            if unique_functions_conf == 'hash':
+                unique_functions = True
+                unique_functions_hash = True
+            elif unique_functions_conf == 'unique_name':
+                unique_functions = True
+                unique_functions_hash = False
+            elif unique_functions_conf == 'none':
+                unique_functions = False
+            else:
+                raise ValueError(
+                    f'Unknown unique_functions configuration: {unique_functions_conf}'
+                )
+            backup = []
+            for node in scope.nodes():
+                if (isinstance(node, nd.NestedSDFG)):
+                    if unique_functions and not unique_functions_hash and node.unique_name != "":
+                        backup.append(
+                            (node, True, copy.deepcopy(node.unique_name),
+                             copy.deepcopy(node.symbol_mapping),
+                             copy.deepcopy(node.sdfg.constants_prop)))
+                        node.unique_name = f"{node.unique_name}_{param}{paramval}"
+                    else:
+                        backup.append(
+                            (node, False, copy.deepcopy(node.sdfg.name),
+                             copy.deepcopy(node.symbol_mapping),
+                             copy.deepcopy(node.sdfg.constants_prop)))
+                        node.sdfg.name = f"{node.sdfg.name}_{param}{paramval}"
```

Comment:
Would it work to just always set _both_ `name` and `unique_name`, regardless of the configuration? It would clean up this whole function quite a lot

Reply:
Theoretically yes, but I'm a bit afraid to do so, because I'm not so sure how unique_name is supposed to be used. However at the moment the only place that uses this at all is the nested sdfg generation, so it's probably fine.